Mathlib/Order/WithBot.lean

```diff
@@ -1122,104 +774,111 @@ theorem strictMono_map_iff {f : α → β} : StrictMono (WithTop.map f) ↔ Stri
 
 alias ⟨_, _root_.StrictMono.withTop_map⟩ := strictMono_map_iff
 
-theorem map_le_iff (f : α → β) (a b : WithTop α)
-    (mono_iff : ∀ {a b}, f a ≤ f b ↔ a ≤ b) :
-    a.map f ≤ b.map f ↔ a ≤ b := by
-  rw [← toDual_le_toDual_iff, toDual_map, toDual_map, WithBot.map_le_iff, toDual_le_toDual_iff]
-  simp [mono_iff]
+theorem map_le_iff (f : α → β) (mono_iff : ∀ {a b}, f a ≤ f b ↔ a ≤ b) :
+    x.map f ≤ y.map f ↔ x ≤ y := by cases x <;> cases y <;> simp [mono_iff]
 
-theorem coe_untop'_le (a : WithTop α) (b : α) : a.untop' b ≤ a :=
-  WithBot.le_coe_unbot' (α := αᵒᵈ) a b
+theorem coe_untop'_le (y : WithTop α) (a : α) : y.untop' a ≤ y :=  by cases y <;> simp
 
 @[simp]
-theorem coe_top_lt [OrderTop α] {x : WithTop α} : (⊤ : α) < x ↔ x = ⊤ :=
-  WithBot.lt_coe_bot (α := αᵒᵈ)
+theorem coe_top_lt [OrderTop α] : (⊤ : α) < x ↔ x = ⊤ := by cases x <;> simp
 
-lemma forall_gt_iff_eq_top {x : WithTop α} : (∀ y : α, y < x) ↔ x = ⊤ :=
-  WithBot.forall_lt_iff_eq_bot (α := αᵒᵈ)
+lemma forall_gt_iff_eq_top : (∀ a : α, a < y) ↔ y = ⊤ := by
+  cases y <;> simp; simpa using ⟨_, lt_irrefl _⟩
 
-lemma forall_ge_iff_eq_top [NoMaxOrder α] {x : WithTop α} : (∀ y : α, y ≤ x) ↔ x = ⊤ :=
-  WithBot.forall_le_iff_eq_bot (α := αᵒᵈ)
+lemma forall_ge_iff_eq_top [NoMaxOrder α] : (∀ a : α, a ≤ y) ↔ y = ⊤ := by
```

Comment:
This proof is also long enough that I question the gain from removing the def-eq.

Reply:
Reverted (for now)